src/carbonio-files-ui-common/views/components/sharing/publicLink/AccessCodeSection.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import React, { useCallback, useState } from 'react';
+
+import {
+	Button,
+	Container,
+	Input,
+	InputProps,
+	Switch,
+	Tooltip,
+	useSnackbar
+} from '@zextras/carbonio-design-system';
+import { useTranslation } from 'react-i18next';
+
+import { copyToClipboard } from '../../../../utils/utils';
+
+interface AccessCodeSectionProps {
+	accessCode: string;
+	isAccessCodeEnabled: boolean;
+	regenerateAccessCode: () => void;
+	toggleAccessCode: () => void;
+}
+
+export const AccessCodeSection = ({
+	accessCode,
+	isAccessCodeEnabled,
+	regenerateAccessCode,
+	toggleAccessCode
+}: AccessCodeSectionProps): React.JSX.Element => {
+	const [t] = useTranslation();
+
+	const [isAccessCodeShown, setIsAccessCodeShown] = useState(false);
+
+	const createSnackbar = useSnackbar();
+
+	const copyAccessCode = useCallback(() => {
+		copyToClipboard(accessCode).then(() => {
+			createSnackbar({
+				key: new Date().toLocaleString(),
+				severity: 'info',
+				label: t('snackbar.clipboard.accessCodeCopied', 'Access code copied'),
+				replace: true,
+				hideButton: true
+			});
+		});
+	}, [createSnackbar, accessCode, t]);
+
+	const toggleShowAccessCode = useCallback(() => {
+		setIsAccessCodeShown((prevState) => !prevState);
+	}, []);
+
+	const CustomElement = useCallback<NonNullable<InputProps['CustomIcon']>>(
+		() => (
+			<Button
+				type={'ghost'}
+				color={'gray0'}
```

Comment:
Just to follow the default of the Icon
```suggestion
				color={'text'}
```